src/SonarScanner.MSBuild.PreProcessor/ProcessedArgs.cs

```diff
@@ -62,6 +64,16 @@ public class ProcessedArgs
         /// </summary>
         public bool InstallLoaderTargets { get; private set; }
 
+        /// <summary>
+        /// Path to the Java executable.
+        /// </summary>
+        public string JavaExePath { get; }
```

Comment:
The logic should be
* Parameter not specified: `null`
* Parameter specified: [File.Exists](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.exists?view=net-8.0) throws or returns false: Error

Reply:
To implement `File.Exists` we should probably modify the image that is running the CI.

Reply:
That is what IFileWrapper is for.